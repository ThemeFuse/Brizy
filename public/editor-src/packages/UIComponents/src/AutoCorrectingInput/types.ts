import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
