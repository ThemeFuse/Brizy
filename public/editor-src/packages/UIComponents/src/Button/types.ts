import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
