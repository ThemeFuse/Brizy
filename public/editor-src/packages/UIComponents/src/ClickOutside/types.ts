import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
