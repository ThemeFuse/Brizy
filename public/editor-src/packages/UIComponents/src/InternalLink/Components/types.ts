import { CSSProperties, ReactElement } from "react";

export interface SelectDropdownProps {
  style?: CSSProperties;
  isSearchLoading?: boolean;
  children: ReactElement;
  onSearchChange?: (s: string) => void;
}

export interface SelectItemProps {
  title: string;
  onClick?: () => void;
}

export interface InternalLinkValueProps {
  className?: string;
  value?: string;
  onClick?: VoidFunction;
  onRemove?: VoidFunction;
}

export interface SearchProps {
  isLoading?: boolean;
  onChange?: (e: string) => void;
}
