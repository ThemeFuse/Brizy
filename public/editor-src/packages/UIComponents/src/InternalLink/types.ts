import { Literal } from "../utils/types/Literal";
import { MValue } from "../utils/value";

export interface Post {
  id: string;
  title: string;
}

export type Choice = {
  icon?: string;
  title: string;
  value: Literal;
};

export enum Status {
  INITIAL = "initial",
  SUCCESS = "success",
  ERROR = "error",
  NO_RESULT = "no_result"
}

export interface Props {
  className?: string;
  value: MValue<Post>;
  placeholder?: string;
  items: Choice[];
  status: Status;
  isLoading?: boolean;
  resetValue: VoidFunction;
  onSearch: (v: string) => void;
  onChange: (v: Choice) => void;
}
