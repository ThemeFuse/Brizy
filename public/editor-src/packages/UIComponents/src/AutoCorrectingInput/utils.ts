import { clamp } from "../utils/math";

export function formatInputValue(value: number, step: number): string | number {
  const stepDecimalLength = decimalLength(step);

  return stepDecimalLength > 0
    ? value.toFixed(stepDecimalLength)
    : String(value);
}

export function decimalLength(number: number): number {
  const [, decimal] = String(number).split(".");
  return decimal ? decimal.length : 0;
}

export function correctNumber(
  number: number,
  min: number,
  max: number,
  step: number
): number {
  return roundByStep(clamp(number, min, max), step, min);
}

export const preciseAdd = preciseOperation.bind(null, "add");

export const preciseSub = preciseOperation.bind(null, "sub");

function roundByStep(number: number, step: number, offset = 0) {
  // this is done to avoid quirks with floating point operations
  // like 1.1 - 1 = 0.10000000000000009
  const mul = 10 ** decimalLength(step);

  number = number * mul;
  step = step * mul;
  offset = offset * mul;

  const tmp = Math.ceil((number - offset) / step) * step + offset;

  return tmp / mul;
}

function preciseOperation(type: "add" | "sub", a: number, b: number) {
  const dla = decimalLength(a);
  const dlb = decimalLength(b);
  const dlMax = Math.max(dla, dlb);
  const mul = 10 ** dlMax;

  const a_ = a * mul;
  const b_ = b * mul;
  let res;

  switch (type) {
    case "add":
      res = a_ + b_;
      break;
    case "sub":
      res = a_ - b_;
      break;
    default:
      throw new Error(`operation type unknown ${type}`);
  }

  return res / mul;
}
