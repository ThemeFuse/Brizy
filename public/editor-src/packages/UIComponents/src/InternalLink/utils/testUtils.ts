import { waitFor } from "@testing-library/react";
import { Choice } from "../types";

export const waitIcon = async (container: HTMLElement) =>
  await waitFor(() =>
    expect(container.querySelector(".brz-icon-svg")).toBeInTheDocument()
  );

export const items: Choice[] = [
  { title: "first", value: 1 },
  { title: "second", value: 2 },
  { title: "third", value: 3 },
  { title: "fourth", value: 4 },
  { title: "fifth", value: 5 },
  { title: "sixth", value: 6 },
  { title: "seventh", value: 7 },
  { title: "eighth", value: 8 },
  { title: "ninth", value: 9 },
  { title: "tenth", value: 10 },
  { title: "eleventh", value: 11 },
  { title: "twelfth", value: 12 },
  { title: "thirteenth", value: 13 },
  { title: "fourteenth", value: 14 },
  { title: "fifteenth", value: 15 }
];
