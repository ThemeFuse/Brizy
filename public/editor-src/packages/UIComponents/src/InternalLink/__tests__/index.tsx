import { fireEvent, render } from "@testing-library/react";
import React from "react";
import _ from "underscore";
import { InternalLink } from "../index";
import { Props, Status } from "../types";
import { items, waitIcon } from "../utils/testUtils";

describe("Test InternalLink", () => {
  const value = { id: items[0].value.toString(), title: items[0].title };

  const component = (props?: Partial<Props>) => (
    <InternalLink
      className="className"
      items={items}
      value={undefined}
      status={Status.INITIAL}
      resetValue={_.noop}
      onChange={_.noop}
      onSearch={_.noop}
      isLoading={false}
      {...props}
    />
  );

  const openDropdown = (container: HTMLElement) => {
    const valueEl = container.querySelector(
      ".brz-ed-control__internalLink__value"
    );
    valueEl && fireEvent.click(valueEl);
  };

  describe("Snapshots and Props", () => {
    test("className should be className", () => {
      const { container } = render(component());

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(".brz-ed-control__internalLink__value")
      ).toHaveClass("className");
    });

    test(`value should be ${value}`, () => {
      const { container } = render(component({ value }));

      const valueSpan = container.querySelector<HTMLSpanElement>(
        ".brz-ed-control__internalLink__value .brz-span"
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(
          ".brz-ed-control__internalLink__value--placeholder"
        )
      ).not.toBeInTheDocument();
      expect(valueSpan?.innerHTML).toBe(value.title);
    });

    test("default Placeholder should be Type name", () => {
      const { container } = render(component());

      const valueSpan = container.querySelector<HTMLSpanElement>(
        ".brz-ed-control__internalLink__value .brz-span"
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(
          ".brz-ed-control__internalLink__value--placeholder"
        )
      ).toBeInTheDocument();
      expect(valueSpan?.innerHTML).toBe("Type name");
    });

    test("placeholder Should be Type value", () => {
      const { container } = render(component({ placeholder: "Type value" }));

      const valueSpan = container.querySelector<HTMLSpanElement>(
        ".brz-ed-control__internalLink__value .brz-span"
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(
          ".brz-ed-control__internalLink__value--placeholder"
        )
      ).toBeInTheDocument();
      expect(valueSpan?.innerHTML).toBe("Type value");
    });

    test("Items should be rendered as list of selectItem", () => {
      const { container } = render(component());

      openDropdown(container);

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelectorAll(".brz-ed-control__internalLink__option")
      ).toHaveLength(items.length);
    });

    test("if Status is no_result should be rendered selectItemNoResults", () => {
      const { container } = render(component({ status: Status.NO_RESULT }));

      openDropdown(container);

      const selectItem = container.querySelector<HTMLLIElement>(
        ".brz-ed-control__multiSelect2__option"
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(selectItem?.innerHTML).toBe("No results");
    });

    test("if Status isn't no_result should be rendered selectItem", () => {
      const { container } = render(component());

      openDropdown(container);

      const selectItem = container.querySelector<HTMLLIElement>(
        ".brz-ed-control__internalLink__option-text"
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(selectItem?.innerHTML).toBe(items[0].title);
    });
    test("if isn't Loading we shouldn't have brz-ed-control__internalLink__dropdown--search-loading className on dropdown", () => {
      const { container } = render(component());

      openDropdown(container);

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(
          ".brz-ed-control__internalLink__dropdown.brz-ed-control__internalLink__dropdown--search-loading"
        )
      ).not.toBeInTheDocument();
    });
    test("if is Loading we should have brz-ed-control__internalLink__dropdown--search-loading className on dropdown", () => {
      const { container } = render(component({ isLoading: true }));

      openDropdown(container);

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(
          ".brz-ed-control__internalLink__dropdown.brz-ed-control__internalLink__dropdown--search-loading"
        )
      ).toBeInTheDocument();
    });
  });
  describe("Event Handlers", () => {
    const resetValue = jest.fn();
    const onSearch = jest.fn();
    const onChange = jest.fn();

    const c = (props?: Partial<Props>) =>
      component({ ...props, resetValue, onSearch, onChange });
    test("Check resetValue", async () => {
      const { container } = render(c({ value }));

      await waitIcon(container);

      const icon = container.querySelector(".brz-icon-svg");

      icon && fireEvent.click(icon);

      expect(resetValue).toHaveBeenCalledTimes(1);
    });
    test("Check onSearch", () => {
      const { container } = render(c());

      openDropdown(container);

      const input = container.querySelector(".brz-input");

      input && fireEvent.change(input, { target: { value: "value" } });

      expect(onSearch).toHaveBeenCalledWith("value");
    });
    describe("Check onChange on every item", () => {
      test.each(items)("onChange Should be called", (item) => {
        const { container, getByTitle } = render(c());

        openDropdown(container);
        fireEvent.click(getByTitle(item.title));

        expect(onChange).toHaveBeenCalledWith(item);
      });
      test(`onChange Should be called ${items.length} times`, () => {
        expect(onChange).toHaveBeenCalledTimes(items.length);
      });
    });
  });
});
