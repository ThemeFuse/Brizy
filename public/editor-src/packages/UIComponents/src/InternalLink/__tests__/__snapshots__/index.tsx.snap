// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InternalLink Snapshots and Props Items should be rendered as list of selectItem 1`] = `
<div
  class="brz-ed-control__internalLink"
>
  <div
    class="brz-ed-control__internalLink__value brz-ed-control__internalLink__value--placeholder className"
    style="position: relative;"
  >
    <span
      class="brz-span"
    >
      Type name
    </span>
  </div>
  <div
    style="position: relative;"
  >
    <div
      class="brz-ed-control__internalLink__dropdown brz-ed-control__internalLink__dropdown--with-search"
      style="position: absolute; left: 0px; top: 0px;"
    >
      <div
        class="brz-ed-control__internalLink__search"
      >
        <input
          class="brz-input"
          placeholder="Type to Search ..."
          spellcheck="false"
        />
      </div>
      <div
        style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 0; max-height: 150px;"
      >
        <div
          style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: 0; max-height: 150px;"
        >
          <ul
            class="brz-ul"
          >
            <li
              class="brz-ed-control__internalLink__option"
              title="first"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                first
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="second"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                second
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="third"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                third
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fourth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fourth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fifth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fifth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="sixth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                sixth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="seventh"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                seventh
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="eighth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                eighth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="ninth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                ninth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="tenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                tenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="eleventh"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                eleventh
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="twelfth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                twelfth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="thirteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                thirteenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fourteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fourteenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fifteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fifteenth
              </span>
            </li>
          </ul>
        </div>
        <div
          style="position: absolute; height: 6px; display: none; right: 2px; bottom: 2px; left: 2px; border-radius: 3px;"
        >
          <div
            style="position: relative; display: block; height: 100%; cursor: pointer; border-radius: inherit; background-color: rgba(0, 0, 0, 0.2);"
          />
        </div>
        <div
          style="position: absolute; width: 6px; display: none; right: 2px; bottom: 2px; top: 2px; border-radius: 3px;"
        >
          <div
            class="brz-ed-control__multiSelect__scroll-thumb"
            style="position: relative; display: block; width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test InternalLink Snapshots and Props className should be className 1`] = `
<div
  class="brz-ed-control__internalLink"
>
  <div
    class="brz-ed-control__internalLink__value brz-ed-control__internalLink__value--placeholder className"
    style="position: relative;"
  >
    <span
      class="brz-span"
    >
      Type name
    </span>
  </div>
  <div
    style="position: relative;"
  />
</div>
`;

exports[`Test InternalLink Snapshots and Props default Placeholder should be Type name 1`] = `
<div
  class="brz-ed-control__internalLink"
>
  <div
    class="brz-ed-control__internalLink__value brz-ed-control__internalLink__value--placeholder className"
    style="position: relative;"
  >
    <span
      class="brz-span"
    >
      Type name
    </span>
  </div>
  <div
    style="position: relative;"
  />
</div>
`;

exports[`Test InternalLink Snapshots and Props if Status is no_result should be rendered selectItemNoResults 1`] = `
<div
  class="brz-ed-control__internalLink"
>
  <div
    class="brz-ed-control__internalLink__value brz-ed-control__internalLink__value--placeholder className"
    style="position: relative;"
  >
    <span
      class="brz-span"
    >
      Type name
    </span>
  </div>
  <div
    style="position: relative;"
  >
    <div
      class="brz-ed-control__internalLink__dropdown brz-ed-control__internalLink__dropdown--with-search"
      style="position: absolute; left: 0px; top: 0px;"
    >
      <div
        class="brz-ed-control__internalLink__search"
      >
        <input
          class="brz-input"
          placeholder="Type to Search ..."
          spellcheck="false"
        />
      </div>
      <div
        style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 0; max-height: 150px;"
      >
        <div
          style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: 0; max-height: 150px;"
        >
          <li
            class="brz-ed-control__multiSelect2__option brz-ed-control__multiSelect2__option--disabled"
          >
            No results
          </li>
        </div>
        <div
          style="position: absolute; height: 6px; display: none; right: 2px; bottom: 2px; left: 2px; border-radius: 3px;"
        >
          <div
            style="position: relative; display: block; height: 100%; cursor: pointer; border-radius: inherit; background-color: rgba(0, 0, 0, 0.2);"
          />
        </div>
        <div
          style="position: absolute; width: 6px; display: none; right: 2px; bottom: 2px; top: 2px; border-radius: 3px;"
        >
          <div
            class="brz-ed-control__multiSelect__scroll-thumb"
            style="position: relative; display: block; width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test InternalLink Snapshots and Props if Status isn't no_result should be rendered selectItem 1`] = `
<div
  class="brz-ed-control__internalLink"
>
  <div
    class="brz-ed-control__internalLink__value brz-ed-control__internalLink__value--placeholder className"
    style="position: relative;"
  >
    <span
      class="brz-span"
    >
      Type name
    </span>
  </div>
  <div
    style="position: relative;"
  >
    <div
      class="brz-ed-control__internalLink__dropdown brz-ed-control__internalLink__dropdown--with-search"
      style="position: absolute; left: 0px; top: 0px;"
    >
      <div
        class="brz-ed-control__internalLink__search"
      >
        <input
          class="brz-input"
          placeholder="Type to Search ..."
          spellcheck="false"
        />
      </div>
      <div
        style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 0; max-height: 150px;"
      >
        <div
          style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: 0; max-height: 150px;"
        >
          <ul
            class="brz-ul"
          >
            <li
              class="brz-ed-control__internalLink__option"
              title="first"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                first
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="second"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                second
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="third"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                third
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fourth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fourth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fifth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fifth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="sixth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                sixth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="seventh"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                seventh
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="eighth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                eighth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="ninth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                ninth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="tenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                tenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="eleventh"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                eleventh
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="twelfth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                twelfth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="thirteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                thirteenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fourteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fourteenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fifteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fifteenth
              </span>
            </li>
          </ul>
        </div>
        <div
          style="position: absolute; height: 6px; display: none; right: 2px; bottom: 2px; left: 2px; border-radius: 3px;"
        >
          <div
            style="position: relative; display: block; height: 100%; cursor: pointer; border-radius: inherit; background-color: rgba(0, 0, 0, 0.2);"
          />
        </div>
        <div
          style="position: absolute; width: 6px; display: none; right: 2px; bottom: 2px; top: 2px; border-radius: 3px;"
        >
          <div
            class="brz-ed-control__multiSelect__scroll-thumb"
            style="position: relative; display: block; width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test InternalLink Snapshots and Props if is Loading we should have brz-ed-control__internalLink__dropdown--search-loading className on dropdown 1`] = `
<div
  class="brz-ed-control__internalLink"
>
  <div
    class="brz-ed-control__internalLink__value brz-ed-control__internalLink__value--placeholder className"
    style="position: relative;"
  >
    <span
      class="brz-span"
    >
      Type name
    </span>
  </div>
  <div
    style="position: relative;"
  >
    <div
      class="brz-ed-control__internalLink__dropdown brz-ed-control__internalLink__dropdown--with-search brz-ed-control__internalLink__dropdown--search-loading"
      style="position: absolute; left: 0px; top: 0px;"
    >
      <div
        class="brz-ed-control__internalLink__search"
      >
        <input
          class="brz-input"
          placeholder="Type to Search ..."
          spellcheck="false"
        />
      </div>
      <div
        style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 0; max-height: 150px;"
      >
        <div
          style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: 0; max-height: 150px;"
        >
          <ul
            class="brz-ul"
          >
            <li
              class="brz-ed-control__internalLink__option"
              title="first"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                first
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="second"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                second
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="third"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                third
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fourth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fourth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fifth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fifth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="sixth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                sixth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="seventh"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                seventh
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="eighth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                eighth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="ninth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                ninth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="tenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                tenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="eleventh"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                eleventh
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="twelfth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                twelfth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="thirteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                thirteenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fourteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fourteenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fifteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fifteenth
              </span>
            </li>
          </ul>
        </div>
        <div
          style="position: absolute; height: 6px; display: none; right: 2px; bottom: 2px; left: 2px; border-radius: 3px;"
        >
          <div
            style="position: relative; display: block; height: 100%; cursor: pointer; border-radius: inherit; background-color: rgba(0, 0, 0, 0.2);"
          />
        </div>
        <div
          style="position: absolute; width: 6px; display: none; right: 2px; bottom: 2px; top: 2px; border-radius: 3px;"
        >
          <div
            class="brz-ed-control__multiSelect__scroll-thumb"
            style="position: relative; display: block; width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test InternalLink Snapshots and Props if isn't Loading we shouldn't have brz-ed-control__internalLink__dropdown--search-loading className on dropdown 1`] = `
<div
  class="brz-ed-control__internalLink"
>
  <div
    class="brz-ed-control__internalLink__value brz-ed-control__internalLink__value--placeholder className"
    style="position: relative;"
  >
    <span
      class="brz-span"
    >
      Type name
    </span>
  </div>
  <div
    style="position: relative;"
  >
    <div
      class="brz-ed-control__internalLink__dropdown brz-ed-control__internalLink__dropdown--with-search"
      style="position: absolute; left: 0px; top: 0px;"
    >
      <div
        class="brz-ed-control__internalLink__search"
      >
        <input
          class="brz-input"
          placeholder="Type to Search ..."
          spellcheck="false"
        />
      </div>
      <div
        style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 0; max-height: 150px;"
      >
        <div
          style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: 0; max-height: 150px;"
        >
          <ul
            class="brz-ul"
          >
            <li
              class="brz-ed-control__internalLink__option"
              title="first"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                first
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="second"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                second
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="third"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                third
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fourth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fourth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fifth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fifth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="sixth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                sixth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="seventh"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                seventh
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="eighth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                eighth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="ninth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                ninth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="tenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                tenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="eleventh"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                eleventh
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="twelfth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                twelfth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="thirteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                thirteenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fourteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fourteenth
              </span>
            </li>
            <li
              class="brz-ed-control__internalLink__option"
              title="fifteenth"
            >
              <span
                class="brz-ed-control__internalLink__option-text"
              >
                fifteenth
              </span>
            </li>
          </ul>
        </div>
        <div
          style="position: absolute; height: 6px; display: none; right: 2px; bottom: 2px; left: 2px; border-radius: 3px;"
        >
          <div
            style="position: relative; display: block; height: 100%; cursor: pointer; border-radius: inherit; background-color: rgba(0, 0, 0, 0.2);"
          />
        </div>
        <div
          style="position: absolute; width: 6px; display: none; right: 2px; bottom: 2px; top: 2px; border-radius: 3px;"
        >
          <div
            class="brz-ed-control__multiSelect__scroll-thumb"
            style="position: relative; display: block; width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test InternalLink Snapshots and Props placeholder Should be Type value 1`] = `
<div
  class="brz-ed-control__internalLink"
>
  <div
    class="brz-ed-control__internalLink__value brz-ed-control__internalLink__value--placeholder className"
    style="position: relative;"
  >
    <span
      class="brz-span"
    >
      Type value
    </span>
  </div>
  <div
    style="position: relative;"
  />
</div>
`;

exports[`Test InternalLink Snapshots and Props value should be [object Object] 1`] = `
<div
  class="brz-ed-control__internalLink"
>
  <div
    class="brz-ed-control__internalLink__value className"
    style="position: relative;"
  >
    <span
      class="brz-span"
    >
      first
    </span>
  </div>
  <div
    style="position: relative;"
  />
</div>
`;
